src/test/java/teammates/test/cases/browsertests/StudentHomePageUiTest.java

```diff
@@ -196,4 +198,9 @@ private void loginWithPersistenceProblem() {
 
     }
 
+    @AfterClass
+    public void classTearDown() {
+        BackDoor.removeDataBundle(testData);
```

Comment:
We should document something like this:
``` Java
// The data associated with the test accounts have to be removed when the test ends because
// the test accounts are shared across tests. In particular, the courses of test student1 account is not the same as
// `StudentHomePageUiTest` so it has to be cleared before running that test. This also implies that other tests that
// runs at the same time may retrieve data created from this test.
//
// Thus, another test running at the same time may read data that was created in this test. This means if the data
// in this test would interfere with another test must run after this test.

// Another implication is that it means that test data is never cleared after. This is to save datastore costs
// as we already remove and restore the data bundle when the test runs.
//
// This raises the question of why we do not remove the data bundle when the class ends then we do not have to remove
// it from startup. The reason is that in the case a test fails to tear down (fail to remove data bundle),
// another test should have no reason to fail.
//
// Lastly, this means that removing the data bundle on startup is not always sufficient because a test only knows how
// to remove its associated data. This is the reason why `StudentHomePageUiTest` would fail if we don't remove the
// data bundle in this test.
// 
// Extending removeDataBundle to remove data outside its associated data would introduce unnecessary complications.
// For example, we don't know really know how much more extra data to remove. Removing too much data would not just
// incur higher datastore costs but we can make tests unexpectedly pass/fail when the data is expected to be
// not present/present in another test.
```

But not all of the comments here should be documented here. Can you extract the relevant parts to more appropriate locations?